src/components/app-single-post/app-single-post.tsx

```diff
@@ -0,0 +1,124 @@
+import { Component, h, Prop, State } from '@stencil/core';
+import { MatchResults, RouterHistory } from '@stencil/router';
+import { BlogPost, BlogPostService } from '../../services/blogPost';
+
+@Component({
+  tag: 'app-single-post',
+  styleUrl: 'app-single-post.scss',
+  shadow: true,
+})
+export class AppSinglePost {
+  @Prop() match: MatchResults;
+  @Prop() history: RouterHistory;
+
+  @State() singleBlogPost: BlogPost;
```

Comment:
variable name `blogPost` indicates single only, no need to write `singleBlogPost`